test/unit/mdc-top-app-bar/standard.foundation.test.js

```diff
@@ -0,0 +1,261 @@
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {assert} from 'chai';
+import td from 'testdouble';
+
+import MDCTopAppBarFoundation from '../../../packages/mdc-top-app-bar/standard/foundation';
+import {numbers} from '../../../packages/mdc-top-app-bar/constants';
+import {createMockRaf} from '../helpers/raf';
+import lolex from 'lolex';
+
+suite('MDCTopAppBarFoundation');
+
+const setupTest = () => {
+  const mockAdapter = td.object(MDCTopAppBarFoundation.defaultAdapter);
+  td.when(mockAdapter.getTopAppBarHeight()).thenReturn(64);
+  td.when(mockAdapter.getViewportScrollY()).thenReturn(0);
+
+  const foundation = new MDCTopAppBarFoundation(mockAdapter);
+
+  return {foundation, mockAdapter};
+};
+
+const createMockHandlers = (foundation, mockAdapter, mockRaf) => {
+  let scrollHandler;
+  let resizeHandler;
+  td.when(mockAdapter.registerScrollHandler(td.matchers.isA(Function))).thenDo((fn) => {
+    scrollHandler = fn;
+  });
+  td.when(mockAdapter.registerResizeHandler(td.matchers.isA(Function))).thenDo((fn) => {
+    resizeHandler = fn;
+  });
+
+  foundation.init();
+  mockRaf.flush();
+  td.reset();
+  return {scrollHandler, resizeHandler};
+};
+
+test('top app bar: listeners is registered on init', () => {
+  const {foundation, mockAdapter} = setupTest();
+  foundation.init();
+  td.verify(mockAdapter.registerScrollHandler(td.matchers.isA(Function)), {times: 1});
+  td.verify(mockAdapter.registerResizeHandler(td.matchers.isA(Function)), {times: 1});
+});
+
+test('listeners removed on destroy', () => {
+  const {foundation, mockAdapter} = setupTest();
+  td.when(mockAdapter.hasClass(MDCTopAppBarFoundation.cssClasses.SHORT_CLASS)).thenReturn(true);
+  foundation.init();
+  foundation.destroy();
+  td.verify(mockAdapter.deregisterScrollHandler(td.matchers.isA(Function)), {times: 1});
+  td.verify(mockAdapter.deregisterResizeHandler(td.matchers.isA(Function)), {times: 1});
+});
+
+test('top app bar scroll: throttledResizeHandler_ updates topAppBarHeight_ if the top app bar height changes', () => {
+  const {foundation, mockAdapter} = setupTest();
+  foundation.init();
+  assert.isTrue(foundation.topAppBarHeight_ === 64);
+  td.when(mockAdapter.getTopAppBarHeight()).thenReturn(56);
+  foundation.throttledResizeHandler_();
+  assert.isTrue(foundation.topAppBarHeight_ === 56);
+});
+
+test('top app bar scroll: throttledResizeHandler_ does not update topAppBarHeight_ if height is the same', () => {
```

Comment:
This test is essentially testing the same thing as the above test. Does this cover another branch that the above does not?

Reply:
No, this is testing two different branches.